src/enftun_xtt.h

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2018 Xaptum, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#ifndef ENFTUN_XTT_H
+#define ENFTUN_XTT_H
+
+#include <xtt.h>
+
+#include "channel.h"
+#include "packet.h"
+
+struct enftun_xtt
+{
+    const char *requested_client_id;
+    const char *server_id;
+    const char *daa_gpk;
+    const char *daa_cred;
+    const char *daa_secretkey;
+    const char *basename;
+    const char *root_cert;
+    const char *client_id;
+    const char *longterm_cert;
+    const char *longterm_private_key;
+    const char *server_ip;  //same as server_host or host
+    const char *server_port;
+    int use_tpm;
+    const char *tcti_dev;
+    const char *suitespec;
+};
+
+typedef enum {
+    XTT_TCTI_SOCKET,
+    XTT_TCTI_DEVICE,
+} xtt_tcti_type;
+
+int xtt_read_from_file(const char *filename, unsigned char *buffer, size_t bytes_to_read);
+int xtt_save_to_file(unsigned char *buffer, size_t bytes_to_write, const char *filename);
+
+int
+enftun_xtt_init(struct enftun_xtt* xtt);
+
+int
+enftun_xtt_handshake(const char *requested_client_id_file, const char *server_id_file,
```

Comment:
Removing support for software DAA should drastically cut the number of arguments here.